File: BE/DTOs/CourseDTO.cs
using BE.ConstanctValue;
using System.ComponentModel.DataAnnotations;

namespace BE.DTOs
{
    public class CourseDTO
    {
        [Required]
        public string Id { get; set; } = "";

        [Required]
        public string Name { get; set; } = "";

        public string? HostName { get; set; } = "";
        public string? HostId { get; set; } = "";

        [Required]
        public bool IsPrivate { get; set; } = true;

        public string? Avatar { get; set; } = "";

        public string? Description { get;  set; } = "";
        public string? StateOfJoining { get; set; } = "";

        public CourseDTO(string id, string name, string hostId, string hostName, bool isPrivate, string? avatar, string? description, string stateOfJoinCourse)
        {
            Id = id;
            Name = name;
            HostName = hostName;
            IsPrivate = isPrivate;
            Avatar = avatar;
            Description = description;
            HostId = hostId;
            StateOfJoining = stateOfJoinCourse;
        }

        public CourseDTO() { }

    }
}

File: FE/app/src/main/java/com/example/tqt_quiz/domain/dto/CourseDTO.java
package com.example.tqt_quiz.domain.dto;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

public class CourseDTO {
    @SerializedName("id")
    @NonNull
    private String id;

    @SerializedName("name")
    @NonNull
    private String name;

    @SerializedName("hostName")
    private String hostName;

    @SerializedName("hostId")
    private String hostId;

    @SerializedName("isPrivate")
    private boolean isPrivate;

    @SerializedName("avatar")
    private String avatar;

    @SerializedName("description")
    private String description;
    @SerializedName("stateOfJoining")
    private String stateOfJoining;

    public String getStateOfJoining() {
        return stateOfJoining;
    }

    public void setStateOfJoining(String stateOfJoining) {
        this.stateOfJoining = stateOfJoining;
    }

    // Getter and Setter for id
    @NonNull
    public String getId() {
        return id;
    }

    public void setId(@NonNull String id) {
        this.id = id;
    }

    // Getter and Setter for name
    @NonNull
    public String getName() {
        return name;
    }

    public void setName(@NonNull String name) {
        this.name = name;
    }

    // Getter and Setter for hostName
    public String getHostName() {
        return hostName;
    }

    public void setHostName(String hostName) {
        this.hostName = hostName;
    }

    // Getter and Setter for isPrivate
    public boolean isPrivate() {
        return isPrivate;
    }

    public void setPrivate(boolean isPrivate) {
        this.isPrivate = isPrivate;
    }

    // Getter and Setter for avatar
    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    // Getter and Setter for description
    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getHostId() {
        return hostId;
    }

    public void setHostId(String hostId) {
        this.hostId = hostId;
    }
}

